homework_4 - Аннотации, Тестовый фрейворк

er3mey opened the pull request:
Чистая версия

er3mey committed:
homework_4 - Аннотации, Тестовый фрейворк

kataus commented on lesson_8_ReflectionAnnotation/src/main/java/ru/sbrf/Runner.java:
```diff
@@ -0,0 +1,61 @@
+        for (Method testMethods : obj.getDeclaredMethods()) {
+            if (testMethods.isAnnotationPresent(Test.class)) {
+                try {
+                    testMethods.invoke(obj.newInstance());
```
Тут у нас получается что несмотря на то что before отработал, но он отработал на другом экземпляре класса. По логике должно быть как:
Для каждого метода тестирования:
1. создаем новый экземпляр
2. Делаем на экземпляре before 
3. Прогоняем на экземпляре метод
4. Делаем на экземпляре after
5. возврат к следующему методу.

kataus commented on lesson_8_ReflectionAnnotation/src/main/java/ru/sbrf/Runner.java:
```diff
@@ -0,0 +1,61 @@
+        int count = 0;
+
+        //Используем название класса-теста для запуска
+        Class<TestExample> obj = TestExample.class;
```
Удобнее такой большой фукнционал делать не в main, а в отдельном классе. тогда мы сможем переиспользовать код

er3mey committed:
homework_4 - done

*Вынес всю логику в класс TestFramework.java
*Создал в TestFramework.java метод runTest(), принимающий на вход класс теста и прогоняющий его
*Поменял логику работы фреймворка, теперь перед и после каждого метода аннотированого @Test прогоняются методы @Before и @After
